tsdb/engine/tsm1/engine.go

```diff
@@ -2124,31 +2174,16 @@ func (e *Engine) compact(wg *sync.WaitGroup) {
 				continue
 			}
 
-			// Find our compaction plans
-			level1Groups, len1 := e.CompactionPlan.PlanLevel(1)
-			level2Groups, len2 := e.CompactionPlan.PlanLevel(2)
-			level3Groups, len3 := e.CompactionPlan.PlanLevel(3)
-			level4Groups, len4 := e.CompactionPlan.Plan(e.LastModified())
-			atomic.StoreInt64(&e.stats.TSMOptimizeCompactionsQueue, len4)
-
-			// If no full compactions are need, see if an optimize is needed
-			var genLen int64
-			if len(level4Groups) == 0 {
-				level4Groups, len4, genLen = e.CompactionPlan.PlanOptimize(e.LastModified())
-				if len(level4Groups) > 0 {
-					for _, group := range level4Groups {
-						e.logger.Info("TSM scheduled for optimized compaction", zap.Strings("files", group))
-					}
+			cycleCount += 1
+			skipOptimize := func() (bool, string) {
+				if cycleCount <= ticksBeforeOptimize {
+					return true, waitMessage
+				} else {
+					return false, ""
```

Comment:
It might be better to do a `startTime := ticks.Now()` before the loop and then do `if time.Since(startTime) > waitForOptimize` here because this loop may not take exactly one `tickPeriod` to run. Also, it seems easier since you don't have to do dimensional analysis to determine if the `ticksBeforeOptimize` calculation is correct.

Reply:
This loop runs indefinitely for the life of the program afaik. Wouldn't start time just always be program start? So after the first time it is greater than waitForOptimize it will always be greater?

Reply:
Yes.  This just avoids optimizing compactions for the first hour after start